### DISCUSSION TITLE
Android Capacitor App in an old Android System WebView.

### DISCUSSION THREAD

**Discussion by valacio** [ionic-team/capacitor#4169] [General]
Hi All, I have this app in React which is compiled for Android using Ionic Capacitor, and I am running into some issues.

It works fine for the Android devices I've tested so far, however when I test it on a Chromebook, I saw some issues, and after some investigation I found out that the Android System WebView in the device is pretty old (68.0.3440.91).

Is there anything I can do to fix this> Maybe overriding the Capacitor Bridge to use Custom Tabs (instead of webView), or force he device to upgrade its WebView when the app is installed/used, or anything else? Otherwise I would need to fix the issues one by one to allow the backward compatibility?

Thanks and regards

**Comment by jcesarmobile** [comment]
Custom Tabs will use the latest version of chrome, but won't have access to the bridge.

Capacitor 2 supports System WebView/Chrome >= 50, Capacitor 3 will support System WebView/Chrome >= 60, so shouldn't have any issues on 68, at least should not be related to Capacitor.

In Capacitor 3, the Device plugin will be a separate package, and the getInfo function will return the System WebView/Chrome version, won't solve your problem, but at least you can prompt the user to update to latest version if their version is old.

**Reply by valacio** [reply]
Hi @jcesarmobile , thanks for your reply.
So far we are not using the Ionic framework (Which I believe is how we would be using the bridge? We are using the capacitor mainly to bundle the React code into the Android app), we also use some java code to support some extra features (Download file, write/read in the device, etc) . Then I believe it would be fine using Custom Tab? But to be honest I don't know if it would be doable to override the Capacitor Bridge to use the Custom Tab instead of Android WebView, what do you think?

The Capacitor 3 is still on Beta, right? Then I believe it wouldn't be safe to use it for production just yet?
Many thanks

**Reply by jcesarmobile** [reply]
It's not required to use Ionic to use the bridge, the bridge allow you to use the Capacitor plugins, the plugins can be used with regular javascript, no framework is required.
If you use Android Custom Tabs, they include a header and toolbar that can't be removed, so your app won't look good.

If you don't use any Capacitor plugin, then you can look at trusted web activities, it's not Capacitor related, more like an alternative that allows to publish Android apps from a progressive web app. But then I don't think you would be able to use the custom java code you said you use to download files.

**Reply by valacio** [reply]
Perfect! I got you, I was already thinking (as a second step in case I was going to use the custom tabs) of how I would get rid of the toolbar, which definitely is something that we don't want to see. Thank you very much.
